doc/man1/openssl-genpkey.pod.in

```diff
@@ -194,18 +194,46 @@ The number of bits in the generated prime. If not specified 2048 is used.
 
 =item B<dsa_paramgen_q_bits>:I<numbits>
 
+=item B<qbits>:I<numbits>
+
 The number of bits in the q parameter. Must be one of 160, 224 or 256. If not
 specified 224 is used.
 
 =item B<dsa_paramgen_md>:I<digest>
 
+=item B<digest>:I<digest>
+
 The digest to use during parameter generation. Must be one of B<sha1>, B<sha224>
 or B<sha256>. If set, then the number of bits in B<q> will match the output size
 of the specified digest and the B<dsa_paramgen_q_bits> parameter will be
 ignored. If not set, then a digest will be used that gives an output matching
 the number of bits in B<q>, i.e. B<sha1> if q length is 160, B<sha224> if it 224
 or B<sha256> if it is 256.
 
+
+=item B<properties>:I<query>
+
+The I<digest> property I<query> string to use when fetching a digest from a provider.
+
+=item B<type>:I<type>
+
+The type of generation to use. Set this to 1 to use legacy FIPS186-2 parameter
+generation. The default of 0 uses FIPS186-4 parameter generation.
+
+=item B<gindex>:I<index>
+
+The index to use for canonical generation and verification of the generator g.
```

Comment:
The values used during generation are the same values used for key validation.
I think what I write here explains what it does.. If you specify the gindex then the generator g can be validated. If not then g is unverifiable.

Reply:
What wasn't entirely clear to me is what a gindex of 1 means vs 2 or 3 or 4 etc. Or is it intended to be a boolean?

Reply:
The algorithm for generation/validation of g requires gindex, seed, e, and p:

```
for (count = 1;  count <= 0xFFFF; ++count) {
    W = Hash(seed || "ggen" || gindex || count)
     g = W^e % p
     if (g >= 2)
        return g;
}
```

Where gindex is just a byte in the range (0..0xFF).
If someone passes a larger value it will just use the bottom byte.

Reply:
Updated the doc..